### PR TITLE
feat(scheduler): expose SharedInformerFactory to the framework handle

### PR DESCRIPTION
**Pull request by draveness**
/kind feature
/sig scheduling
/priority important-soon

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

ref: #83523

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Expose SharedInformerFactory in the framework handle
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: cmd/kube-scheduler/app/server.go [hunk 1]
```diff
@@ -168,6 +168,7 @@ func Run(cc schedulerserverconfig.CompletedConfig, stopCh <-chan struct{}, regis
 
 	// Create the scheduler.
 	sched, err := scheduler.New(cc.Client,
+		cc.InformerFactory,
```
Comment (mrkm4ntr) [inline cmd/kube-scheduler/app/server.go]: I think it is better to pass a different informerFactory. InformerFactory has a cache of informers internally, so this implementation expose informers that scheduler uses.
https://github.com/kubernetes/client-go/blob/8248d0a0e61a6cc77b53557918964cd1038d511f/informers/factory.go#L162-L183

Comment (draveness) [reply cmd/kube-scheduler/app/server.go]: Sorry, what are you suggesting?

Comment (mrkm4ntr) [reply cmd/kube-scheduler/app/server.go]: I was worried that plugin implementers changes lister's cache by mistake and it causes unexpected behavior of scheduler core. But maybe I'm worried too much.

Comment (ahg-g) [reply cmd/kube-scheduler/app/server.go]: since we are passing the sharedInformerFactory, lets remove all listers that are derived from it.

Comment (draveness) [reply cmd/kube-scheduler/app/server.go]: Sure I’ll update the PR

**Comment by ahg-g** [comment]
Thanks please squash.

/approve
/assign @msau42 for volume-related tests

**Comment by draveness** [comment]
> Thanks please squash.

done

**Comment by ahg-g** [comment]
> @draveness: The following test **failed**, say `/retest` to rerun them all:
> 
> Test name	Commit	Details	Rerun command
> pull-kubernetes-integration	[d7db0e2](https://github.com/kubernetes/kubernetes/commit/d7db0e245dd0e2b3a3316a26cf988aa7263f4210)	[link](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/83663/pull-kubernetes-integration/1182294283093282819/)	`/test pull-kubernetes-integration`
> [Full PR test history](https://prow.k8s.io/pr-history?org=kubernetes&repo=kubernetes&pr=83663). [Your PR dashboard](https://gubernator.k8s.io/pr/draveness). Please help us cut down on flakes by [linking to](https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/kubernetes/kubernetes/issues?q=is:issue+is:open) when you hit one in your PR.

this seems like a legitimate failure, we should look closely into it.

**Comment by msau42** [comment]
/approve
/lgtm cancel
to look into test failures

**Comment by draveness** [comment]
> /approve
> /lgtm cancel
> to look into test failures

https://prow.k8s.io/view/gcs/kubernetes-jenkins/logs/ci-kubernetes-integration-master/1179256637836234754

https://testgrid.k8s.io/google-unit#integration-master&include-filter-by-regex=NodePIDPre

It was a flaky test which fails since 10-02 when this PR wasn't opened. Please take another look, the previous approve does not work as expected.

**Pull request merged**